packages/shared/ReactTypes.js

```diff
@@ -70,6 +76,30 @@ export type ReactContext<T> = {
   ...
 };
 
+export type ServerContextJSONValue =
+  | string
+  | boolean
+  | number
+  | null
+  | $ReadOnlyArray<ServerContextJSONValue>
+  | {+[key: string]: ServerContextJSONValue};
+
+export type ReactServerContext<T: ServerContextJSONValue> = {
```

Comment:
It would be good if this could just be:

```js
export type ReactServerContext<T: ServerContextJSONValue> = ReactContext<T>;
```

Because if we are now able to reuse all the internals of how Fiber works with Context already. Then all those calls will try to access these fields. It's nicer to JS VMs if those accesses are only done against one hidden class and so it's nice to be able to reuse the same exact shape of these objects.

That would mean that we could add a field to regular context that ends up unused on those. Another option is to repurpose another field for that.

Reply:
I think you could use `_defaultValue === DEFAULT_PLACEHOLDER` as an indicator for whether it has been initialized yet or not.